server/api/resolvers/requestGroupResolvers.ts

```diff
@@ -83,9 +83,34 @@ const requestGroupMutationResolvers = {
     },
     deleteRequestGroup: async (_, { _id }, { authenticateUser }): Promise<RequestGroupInterface> => {
         return authenticateUser().then(async () => {
-            const requestGroup = await RequestGroup.findById(_id);
-            requestGroup.deletedAt = new Date();
-            return requestGroup.save();
+            return sessionHandler(async (session) => {
+                const requestGroup = await RequestGroup.findById(_id).session(session);
+
+                if (requestGroup.deletedAt != null) return requestGroup.save({ session: session });
+
+                requestGroup.deletedAt = new Date();
+
+                for (const requestTypeEmbedding of requestGroup.requestTypes) {
+                    const requestType = await RequestType.findById(requestTypeEmbedding._id).session(session);
+
+                    if (requestType.deletedAt != null) continue;
+
+                    requestType.deletedAt = new Date();
+                    await requestType.save({ session: session });
+
+                    for (var i = 0; i < requestType.requests.length; i++) {
```

Comment:
Is there a reason why we have a index-based for loop here while the outer loop is not

Reply:
I needed the index to update the embedding (which contains deletedAt) for requestTypes -> requests but since the embedding of requestTypes in requestGroups does not contain deletedAt, I don't need to update that embedding.